TestCachePurging.testBackgroundPurges test failure

### Description

This is a test failure from the `TestCachePurging.testBackgroundPurges` method

```
java.lang.AssertionError: expected:<-1> but was:<483790876047>
    at __randomizedtesting.SeedInfo.seed([EB3E8668829B996D:FB85C0A3D2BC042D]:0)	
    at org.junit.Assert.fail(Assert.java:89)	
    at org.junit.Assert.failNotEquals(Assert.java:835)	
    at org.junit.Assert.assertEquals(Assert.java:647)	
    at org.junit.Assert.assertEquals(Assert.java:633)	
    at org.apache.lucene.monitor.TestCachePurging.testBackgroundPurges(TestCachePurging.java:140)	
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)	
    at java.lang.reflect.Method.invoke(Method.java:565)	
    at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1763)	
    at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)	
    at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)	
    at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)	
    at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)	
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)	
    at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)	
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)	
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)	
    at org.junit.rules.RunRules.evaluate(RunRules.java:20)	
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)	
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)	
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)	
    at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)	
    at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)	
    at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)	
    at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)	
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)	
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
    at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)	
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)	
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)	
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
    at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:52)	
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)	
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)	
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)	
    at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)	
    at org.junit.rules.RunRules.evaluate(RunRules.java:20)	
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)	
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)	
    at java.lang.Thread.run(Thread.java:1474)
```

I was not able to reproduce this with the seed that it failed with, but my assumption is the following:

The `Monitor` class ctor calls the `ReadonlyQueryIndex/WritableQueryIndex`, either of which creates a single threaded scheduled executor and a job at scheduled fixed rate. If there is enough time [between object creation and the following assertion check](https://github.com/apache/lucene/blob/main/lucene/monitor/src/test/org/apache/lucene/monitor/TestCachePurging.java#L136-L140), that tries to ensure that no run of that job has happened so far, the assertion fails.

I am not sure if it's worth keeping that assertion, when one cannot control the execution of that background purging job because it's defined in the Monitor/QueryIndex ctors (i.e. having an explicit start method instead).

### Gradle command to reproduce

```
./gradlew :lucene:monitor:test --tests "org.apache.lucene.monitor.TestCachePurging.testBackgroundPurges" -Ptests.asserts=true -Ptests.file.encoding=US-ASCII -Ptests.gui=true -Ptests.jvmargs= -Ptests.jvms=12 -Ptests.seed=EB3E8668829B996D -Ptests.vectorsize=128
```

--- a/lucene/monitor/src/test/org/apache/lucene/monitor/TestCachePurging.java
+++ b/lucene/monitor/src/test/org/apache/lucene/monitor/TestCachePurging.java
@@ -137,8 +137,6 @@ public class TestCachePurging extends MonitorTestBase {
         new MonitorConfiguration().setPurgeFrequency(50, TimeUnit.MILLISECONDS);
     try (Monitor monitor = new Monitor(ANALYZER, Presearcher.NO_FILTERING, config)) {
 
-      assertEquals(-1, monitor.getQueryCacheStats().lastPurged());
-
       for (int i = 0; i < 100; i++) {
         monitor.register(newMonitorQuery(i));
       }

--- a/lucene/monitor/src/test/org/apache/lucene/monitor/TestMonitor.java
+++ b/lucene/monitor/src/test/org/apache/lucene/monitor/TestMonitor.java
@@ -253,4 +253,26 @@ public class TestMonitor extends MonitorTestBase {
       assertEquals(0, matches.getMatchCount());
     }
   }
+
+  public void testQueryCacheStats() throws IOException {
+    try (Monitor monitor = newMonitor()) {
+      assertEquals(-1, monitor.getQueryCacheStats().lastPurged());
+      assertEquals(0, monitor.getQueryCacheStats().queries());
+      assertEquals(0, monitor.getQueryCacheStats().cachedQueries());
+
+      MonitorQuery[] queries =
+          new MonitorQuery[] {
+            new MonitorQuery("1", parse("test1")),
+            new MonitorQuery("2", parse("test2")),
+            new MonitorQuery("3", parse("test3"))
+          };
+      monitor.register(queries);
+      assertEquals(3, monitor.getQueryCacheStats().queries());
+      assertEquals(3, monitor.getQueryCacheStats().cachedQueries());
+
+      assertEquals(-1, monitor.getQueryCacheStats().lastPurged());
+      monitor.purgeCache();
+      assertTrue(monitor.getQueryCacheStats().lastPurged() > 0);
+    }
+  }
 }

